actionpack/test/controller/renderers_test.rb

```diff
@@ -56,16 +60,20 @@ def setup
     @controller.logger = ActiveSupport::Logger.new(nil)
   end
 
-  def test_using_custom_render_option
+  def test_using_custom_render_format
+    ActionController::Renderers.add_serializer :simon do |says, options|
+      "Simon says: #{says}"
+    end
     ActionController.add_renderer :simon do |says, options|
       self.content_type  = Mime[:text]
-      self.response_body = "Simon says: #{says}"
+      says
     end
 
     get :render_simon_says
     assert_equal "Simon says: foo", @response.body
   ensure
     ActionController.remove_renderer :simon
+    ActionController.remove_serializer :simon
   end
 
   def test_raises_missing_template_no_renderer
```

Comment:
PR NOTE: Just some comments about MissingRenderer perhaps for a future PR

I don't think there's really an easy way to do implement `MissingRenderer` at this time. Given
this file defines


```ruby
def render_to_body(options)
  _render_to_body_with_renderer(options) || super
end
```

When there's no registered renderer,`_render_to_body_with_renderer(options)`
is `nil` and the method calls `super`, which may raise
`ActionView::MissingTemplate` in an ActionController::Base, but just be `nil`
in an ActionController::Metal. So, if we consider a renderer an `API`
concern, in the language of the [`Responders`
gem](https://github.com/plataformatec/responders/blob/f0e093d01e799d8b67952c013daf8808e9ea02db/lib/action_controller/responder.rb#L209),
how do we know which `content_type` or `Mime::Type.lookup(content_type)`
should be a missing template (html/plain/text) vs. missing renderer
(json/xml/js/csv)?

Responders implementation for reference:

```ruby
  # This is the common behavior for formats associated with APIs, such as :xml and :json.
  def api_behavior
    raise MissingRenderer.new(format) unless has_renderer?
  end

  # Check whether the necessary Renderer is available
  def has_renderer?
    Renderers::RENDERERS.include?(format)
  end
```